db.go

```diff
@@ -9,6 +9,8 @@ import (
 	"go.uber.org/multierr"
 )
 
+var backgroundCtx = context.Background() //reduces unnecessary cross library calls
```

Comment:
no, let's not use this. 
this makes backgroundCtx become global. As you know context contains states, and can be canceled. if one is cancelled, all context will be cancelled

Reply:
![Uploading image.png…]()

Reply:
Inside of context.Background()

<img width="228" alt="image" src="https://github.com/bxcodec/dbresolver/assets/24226219/630d0c3c-a6c3-42e7-824f-0d981a1cd791">


Its just returning a global variable

Reply:
This ctx cannot be cancelled or nor is it susceptible to cascading cancelations